data_steward/cdr_cleaner/cleaning_rules/remove_non_matching_participant.py

```diff
@@ -35,214 +45,275 @@
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id,
+                 ehr_dataset_id, validation_dataset_id):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+
+        self.ehr_dataset_id = ehr_dataset_id
+        self.validation_dataset_id = validation_dataset_id
+
+        desc = 'Removes non-matching and not validated participant records from the combined dataset.'
+        super().__init__(issue_numbers=['DC468', 'DC823'],
+                         description=desc,
+                         affected_datasets=[cdr_consts.COMBINED],
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id)
+
+        self.client = BigQueryClient(self.project_id)
+
+    def setup_rule(self, client) -> None:
+        pass
+
+    def setup_validation(self, client) -> None:
+        pass
+
+    def validate_rule(self, client) -> None:
+        pass
+
+    def get_sandbox_tablenames(self) -> list:
+        """
+        Return a list table names created to backup deleted data.
+        """
+        return [
+            f'{table}_{TICKET_NUMBER}'
+            for table in remove_pids.get_tables_with_person_id(
+                self.project_id, self.dataset_id)
+        ]
+
+    def exist_participant_match(self, ehr_dataset_id, hpo_id) -> bool:
+        """
+        Checks if the hpo has submitted the participant_match table or not.
+
+        :param ehr_dataset_id: Unioned EHR dataset ID.
+        :param hpo_id: HPO site ID.
+        :return: True if participant match table exists. Otherwise, False.
+        """
+        return bq_utils.table_exists(
+            bq_utils.get_table_id(hpo_id, PARTICIPANT_MATCH), ehr_dataset_id)
+
+    def exist_identity_match(self, table_id) -> bool:
+        """
+        Checks if the hpo has valid the identity_match table or not.
+
+        :param table_id: ID of the identity_match table.
+        :return: True if identity match table exists. Otherwise, False.
+        """
+        try:
+            self.client.get_table(table_id)
+            return True
+        except NotFound:
+            return False
+
+    def get_missing_criterion(self, field_names) -> str:
+        """
+        Generates a bigquery column expression for missing criteria.
+
+        :param field_names: a list of field names for counting `missing`s
+        :return: Multiple CAST statements joined by ' + '.
+        """
+        joined_column_expr = ' + '.join([
+            CAST_MISSING_COLUMN.render(column=field_name)
+            for field_name in field_names
+        ])
+        return joined_column_expr
+
+    def get_non_match_participants(self,
```

Comment:
Couple of changes here: 
1. Remove all instances of bq_utils (being deprecated) and use client as a passed parameter that has these built in, so you can just call them as methods.
2. Call this function from inside setup_rule since that provides you the client.
3. Ideally you don't need the actual pids as a list of numbers. Use a common table expression to do something like 
`WHERE person_id IN (SELECT person_id FROM table WHERE match_criteria)`